src/lib/application.ts

```diff
@@ -238,9 +238,23 @@ export class Application extends ChildableComponent<Application, AbstractCompone
      * @returns  The list of input files with expanded directories.
      */
     public expandInputFiles(inputFiles?: string[]): string[] {
-        let exclude: IMinimatch, files: string[] = [];
+        let exclude: Array<IMinimatch>, files: string[] = [];
         if (this.exclude) {
-            exclude = new Minimatch(this.exclude);
+            exclude = this.exclude.map(pattern => new Minimatch(pattern));
+        }
+
+        function isExcluded(fileName: string): boolean {
+            if (!exclude) {
+                return false;
+            }
+
+            for (let mm of exclude) {
```

Comment:
This should be a simple one-liner, `return exclude.some(x => x.match(fileName))`.